### PR TITLE
PLANNER-1899,PLANNER-1902: Persist vehicles

### LINKED ISSUE #194: Persist vehicles
Follows up on #144.

### LINKED ISSUE #195: Include vehicles in YAML export/import
Follows up on #144.

### PR DESCRIPTION
**Pull request by yurloc**
Fixes #194 and fixes #195.

### CODE REVIEW

File: optaweb-vehicle-routing-backend/src/main/java/org/optaweb/vehiclerouting/service/vehicle/VehicleRepository.java [hunk 1]
```diff
@@ -20,19 +20,26 @@
...
 
 /**
  * Defines repository operations on vehicles.
  */
 public interface VehicleRepository {
 
     /**
-     * Create a vehicle with a unique ID.
-     * @param name vehicle name
+     * Create vehicle with a unique ID.
```
Comment (michaltomco) [inline optaweb-vehicle-routing-backend/src/main/java/org/optaweb/vehiclerouting/service/vehicle/VehicleRepository.java:31]: comment nitpick: I believe removing the "a" article before "vehicle" is not grammatically correct.

File: optaweb-vehicle-routing-backend/src/main/java/org/optaweb/vehiclerouting/domain/VehicleFactory.java [hunk 2]
```diff
@@ -25,6 +25,16 @@ private VehicleFactory() {
...
+     * @param name vehicle name
+     * @param capacity vehicle capacity
+     * @return vehicle data
+     */
+    public static VehicleData vehicleData(String name, int capacity) {
+        return new VehicleData(name, capacity);
+    }
+
     /**
      * Create a new vehicle with give id, name and capacity.
```
[the remark this answers is not in the document]
Comment (yurloc) [reply optaweb-vehicle-routing-backend/src/main/java/org/optaweb/vehiclerouting/domain/VehicleFactory.java:39]: Yes, absolutely. Thanks!

File: optaweb-vehicle-routing-backend/src/main/java/org/optaweb/vehiclerouting/domain/VehicleFactory.java [hunk 3]
```diff
@@ -25,6 +25,16 @@ private VehicleFactory() {
         throw new AssertionError("Utility class");
     }
 
+    /**
+     * Create vehicle data.
+     * @param name vehicle name
+     * @param capacity vehicle capacity
```
Comment (yurloc) [inline optaweb-vehicle-routing-backend/src/main/java/org/optaweb/vehiclerouting/domain/VehicleFactory.java:31]: @michaltomco Michal, can you help me with this... "vehicle's name/capacity/ID" or just "vehicle name/capacity/ID" without the possessive form? Are both forms correct? Do you prefer one of them?

Comment (michaltomco) [reply optaweb-vehicle-routing-backend/src/main/java/org/optaweb/vehiclerouting/domain/VehicleFactory.java:31]: I believe both are correct, even though "vehicle name" sounds more natural to me (and compact).

**Pull request merged**